### DISCUSSION TITLE
Adding mocap body on rope

### DISCUSSION THREAD

**Discussion by matansudry** [google-deepmind/mujoco#171] [Asking for Help]
Hi,

I am trying to add a mocap body on each rope link.
For step one, I tried to add mocap body to some link, see implementation below.
I am not sure why but the link and the mocap body are not connected.

In the end, I want to give some trajectory with [x,y,z] points and move the rope using the mocap body.

<mujoco model="rope">
    <include file="scene_v2.xml"/>

    <option timestep="0.002" jacobian="dense"/>

    <size nconmax="100" njmax="300" nstack="50000"/>

    <equality>
        <connect body1="hand" body2="B10" anchor="0 0 0" solimp="0.9 0.95 0.001" solref="0.02 1"/>
    </equality>
    <worldbody>
        <body name="B10" pos="0 -0.5 0.00">
            <freejoint/>
            <composite type="rope" count="21 1 1" spacing="0.04" offset="0 0 2">
                <joint kind="main" damping="0.005"/>
                <geom  type="capsule" size=".01 .015" rgba=".8 .2 .1 1"/>
            </composite>
        </body>
        <body name="hand" pos="0 -0.5 0.00" mocap="true">
            <geom type="box" size=".01 .01 .01" rgba="0 .9 0 .5" contype="2"/>
        </body>
    </worldbody>
</mujoco>

**Comment by rohit-kumar-j** [comment]
@matansudry, you might want to check this out: https://github.com/kevincheng3/mocap-control

**Comment by wangjunyi9999** [comment]
Hi, bro, have you found a good solution to it? I also meet the same thing.
@matansudry
